From: Anland Hyprland Maintainers
Date: Fri, 8 Aug 2026 10:45:00 +0800
Subject: [PATCH] compat: log raw color-management weak surface pointers

Forky's hyprutils does not allow a CWeakPointer to be cast directly to an
integer. The trace messages only need the raw address, which get() provides.

--- a/src/protocols/ColorManagement.cpp
+++ b/src/protocols/ColorManagement.cpp
@@ -271,9 +271,9 @@ CColorManagementSurface::CColorManagementSurface(SP<CWpColorManagementSurfaceV1
 
     m_resource->setDestroy([this](CWpColorManagementSurfaceV1* r) {
-        LOGM(Log::TRACE, "Destroy wp cm surface {}", (uintptr_t)m_surface);
+        LOGM(Log::TRACE, "Destroy wp cm surface {}", (uintptr_t)m_surface.get());
         PROTO::colorManagement->destroyResource(this);
     });
     m_resource->setOnDestroy([this](CWpColorManagementSurfaceV1* r) {
-        LOGM(Log::TRACE, "Destroy wp cm surface {}", (uintptr_t)m_surface);
+        LOGM(Log::TRACE, "Destroy wp cm surface {}", (uintptr_t)m_surface.get());
         PROTO::colorManagement->destroyResource(this);
     });
@@ -372,9 +372,9 @@ CColorManagementFeedbackSurface::CColorManagementFeedbackSurface(SP<CWpColorMan
 
     m_resource->setDestroy([this](CWpColorManagementSurfaceFeedbackV1* r) {
-        LOGM(Log::TRACE, "Destroy wp cm feedback surface {}", (uintptr_t)m_surface);
+        LOGM(Log::TRACE, "Destroy wp cm feedback surface {}", (uintptr_t)m_surface.get());
         PROTO::colorManagement->destroyResource(this);
     });
     m_resource->setOnDestroy([this](CWpColorManagementSurfaceFeedbackV1* r) {
-        LOGM(Log::TRACE, "Destroy wp cm feedback surface {}", (uintptr_t)m_surface);
+        LOGM(Log::TRACE, "Destroy wp cm feedback surface {}", (uintptr_t)m_surface.get());
         PROTO::colorManagement->destroyResource(this);
     });